### PR TITLE
chore(libnpmdiff): added as workspace

### PR DESCRIPTION
**Pull request by ruyadorno**
- Setup `./packages/*` as workspaces
- Moved source from: https://github.com/npm/libnpmdiff to `./packages/libnpmdiff`
- Added CI target for running workspaces tests

## References
Relates to: https://github.com/npm/statusboard/issues/362

### CODE REVIEW

**Comment by ruyadorno** [comment]
I wonder if we need to adjust something after https://github.com/npm/cli/pull/3365 lands 🤔 maybe add `./packages` to `"files"` list is going to be necessary?

File: scripts/bundle-and-gitignore-deps.js [hunk 1]
```diff
@@ -10,7 +10,10 @@ const shouldIgnore = []
 
 arb.loadVirtual().then(tree => {
   for (const node of tree.children.values()) {
-    if (node.dev || node.isLink) {
+    const has = (obj, key) => Object.prototype.hasOwnProperty.call(obj, key)
+    const nonProdWorkspace =
+      node.isWorkspace && !(has(tree.package.dependencies, node.name))
```
Comment (ruyadorno) [inline scripts/bundle-and-gitignore-deps.js:15]: @isaacs this makes me think we should have a better API in Arborist to tell the difference between a prod/dev/peer declared workspace vs a workspace-declared-only 🤔

**Comment by wraithgar** [comment]
Fixed one merge conflict, combining the `isLink` check added by moving docs to workspaces with the workspace check this one does in bundle-and-gitignore-deps.  However there is another conflict that only happens locally when trying to merge this, and that is because of the now missing readme/changelog files in node_modules. This PR was made with those present and there's something that happens locally causing a merge conflict, making our pull script fail.

**Comment by ruyadorno** [comment]
@wraithgar rebased it!

**Comment by ruyadorno** [comment]
cool, with this we may archive https://github.com/npm/libnpmdiff

**Pull request merged**